packages/docs-markdown/src/controllers/links/linkToDocsPageByUrl.ts

```diff
@@ -103,27 +105,76 @@ async function getLocalRepoFileLink(
 		if (metadataTags.length > 0) {
 			const metadata = metadataTags[0].getAttribute('content');
 			const repoName = folderPath.split('\\').pop();
-			if (checkIfCurrentRepoIsUrl(metadata, repoName)) {
-				const absoluteFilePath = parseMetadata(metadata);
-				const lastIndex = currentFilePath.lastIndexOf('\\');
-				const currentFilePathDirectory = currentFilePath.substring(0, lastIndex);
-				if (absoluteFilePath) {
-					const pathToLinkFile = join(folderPath, absoluteFilePath);
-					const relativePath = tryGetRelativePath(currentFilePathDirectory, pathToLinkFile);
-					if (relativePath) {
-						if (!altText) {
-							altText = await tryGetHeader(pathToLinkFile);
-						}
-						return { confirmation: '', link: externalLinkBuilder(relativePath, altText) };
+			if (checkIfUrlRepoIsInCurrentRepo(metadata, repoName)) {
+				const content = getDocfxContent(editor);
+				if (
+					content &&
```

Comment:
Is it an option to use ? notation here for your null checks?
`content?.build?.content?.some(file => file.src)`